src/app/loadout/Loadout.tsx

```diff
@@ -0,0 +1,127 @@
+import _ from 'lodash';
+import React, { ReactNode, useMemo, useState } from 'react';
+import ReactDOM from 'react-dom';
+import { useSelector } from 'react-redux';
+import ItemCategorySection from './loadout-ui/ItemCategorySection';
+import PlugDef from './loadout-ui/PlugDef';
+import Subclass from './loadout-ui/Subclass';
+import styles from './Loadout.m.scss';
+import ModAssignmentDrawer from './mod-assignment-drawer/ModAssignmentDrawer';
+import { createGetModRenderKey } from './mod-utils';
+
+export default function Loadout({
+  loadout,
+  store,
+  actionButtons,
+}: {
+  loadout: Loadout;
+  store: DimStore;
+  actionButtons: ReactNode[];
+}) {
+  const defs = useD2Definitions()!;
+  const buckets = useSelector(bucketsSelector)!;
+  const allItems = useSelector(allItemsSelector);
+  const getModRenderKey = createGetModRenderKey();
+  const [showModAssignmentDrawer, setShowModAssignmentDrawer] = useState(false);
+
+  // Turn loadout items into real DimItems, filtering out unequippable items
+  const [items, subclass, warnitems] = useMemo(() => {
+    const [items, warnitems] = getItemsFromLoadoutItems(loadout.items, defs, buckets, allItems);
+    let equippableItems = items.filter((i) => itemCanBeEquippedBy(i, store, true));
+    const subclass = equippableItems.find((i) => i.bucket.hash === BucketHashes.Subclass);
+    if (subclass) {
+      equippableItems = equippableItems.filter((i) => i !== subclass);
+    }
+    return [equippableItems, subclass, warnitems];
+  }, [loadout.items, defs, buckets, allItems, store]);
+
+  const savedMods = getModsFromLoadout(defs, loadout);
+  const equippedItemIds = new Set(loadout.items.filter((i) => i.equipped).map((i) => i.id));
+
+  const categories = _.groupBy(items.concat(warnitems), (i) => i.bucket.sort);
+
+  const showPower = categories.Weapons?.length === 3 && categories.Armor?.length === 5;
+  const power = showPower
+    ? Math.floor(getLight(store, [...categories.Weapons, ...categories.Armor]))
+    : 0;
+
+  return (
+    <div className={styles.loadout} id={loadout.id}>
+      <div className={styles.title}>
+        <h2>
+          <ClassIcon className={styles.classIcon} classType={loadout.classType} />
+          {loadout.name}
+          {warnitems.length > 0 && (
+            <span className={styles.missingItems}>
+              <AppIcon className="warning-icon" icon={faExclamationTriangle} />
+              {t('Loadouts.MissingItemsWarning')}
+            </span>
+          )}
+        </h2>
+        <div className={styles.actions}>{actionButtons}</div>
+      </div>
+      {loadout.notes && <div className={styles.loadoutNotes}>{loadout.notes}</div>}
+      <div className={styles.contents}>
+        {(items.length > 0 || subclass || savedMods.length > 0) && (
+          <>
+            <div>
+              <Subclass defs={defs} subclass={subclass} power={power} />
+            </div>
+            {['Weapons', 'Armor', 'General'].map((category) => (
+              <ItemCategorySection
```

Comment:
Was this already broken out somewhere?

Reply:
Yeah this PR merges into my other one I had up. I just split the changes across PR's to make the diffs more readable